Assets/HoloToolkit/Input/Scripts/Interactable.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Base class Interactable GameObjects can inherit from.
+    /// </summary>
+    public class Interactable : MonoBehaviour
+    {
+        protected bool TriggerKeywordOnGazeOnly;
+
+        protected bool IsFocused;
+
+        protected virtual void OnEnable()
+        {
+            GestureManager.Instance.OnTap += OnTap;
+            GazeManager.Instance.OnGazeExit += OnGazeExit;
+            GazeManager.Instance.OnGazeEnter += OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized += KeywordRecognized;
+        }
+
+        protected virtual void OnDisable()
+        {
+            GestureManager.Instance.OnTap -= OnTap;
+            GazeManager.Instance.OnGazeExit -= OnGazeExit;
+            GazeManager.Instance.OnGazeEnter -= OnGazeEnter;
+            KeywordManager.Instance.OnKeywordRecognized -= KeywordRecognized;
+        }
+
+        /// <summary>
+        /// Called when a user has tapped any gameObject.
+        /// </summary>
+        /// <param name="tappedGameObject">GameObject user has tapped.</param>
+        protected virtual void OnTap(GameObject tappedGameObject)
+        {
+            if (tappedGameObject == gameObject)
```

Comment:
strange I got wildly different results:

Number of Arguments, Number of Listeners, C# Event Time(ms), UnityEvent Time(ms)
0,1,243,3932
0,2,746,6068
0,3,1626,8613
0,4,2096,10055
0,5,3447,12109
0,6,3170,13985
0,7,3593,15937
0,8,4091,17831
0,9,4616,19753
0,10,5729,22745
1,1,244,5852
1,2,878,8165
1,3,1663,10930
1,4,2362,12918
1,5,3230,15817
1,6,2964,17036
1,7,4214,19493
1,8,3866,21435
1,9,4427,23656
1,10,4858,25973
2,1,250,7247
2,2,1002,9600
2,3,1687,12096
2,4,2023,14537
2,5,2425,17074
2,6,2924,19558
2,7,3579,24862
2,8,4425,24441
2,9,4394,25374
2,10,5015,28224